doxia-core/src/main/java/org/apache/maven/doxia/parser/AbstractXmlParser.java

```diff
@@ -194,6 +215,7 @@ protected SinkEventAttributeSet getAttributesFromParser(XmlPullParser parser) {
      * @throws org.apache.maven.doxia.macro.MacroExecutionException if there's a problem executing a macro
      */
     private void parseXml(XmlPullParser parser, Sink sink) throws XmlPullParserException, MacroExecutionException {
+        sink.setDocumentLocator(new XmlPullParserLocator(parser, null));
```

Comment:
@michael-o Is the 3rd argument of https://maven.apache.org/doxia/doxia/doxia-core/apidocs/org/apache/maven/doxia/parser/Parser.html#parse-java.io.Reader-org.apache.maven.doxia.sink.Sink-java.lang.String- supposed to contain the file path? The javadoc is a bit fuzzy here....

Reply:
https://issues.apache.org/jira/browse/DOXIA-224

Reply:
See: https://issues.apache.org/jira/browse/DOXIA-224 and https://github.com/apache/maven-doxia/commit/70642f0fd1dbc7d0fb7780d347b87e1fd9b2d1b0. It seems to be a logical reference omitting the format. I think it is similar to `generator` and `document`: https://github.com/apache/maven-doxia-sitetools/blob/5fe4a4c5359e6a23b78f385e15f77767cadaee99/doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DocumentRenderingContext.java#L66

See how I have been using it throughout.

Reply:
You cannot and must not assume that a parse consumes from a file, it can be any source satisfying the `Reader` interface.